internal/jobs/launch_instance_aws.go

```diff
@@ -212,3 +218,50 @@ func DoLaunchInstanceAWS(ctx context.Context, args *LaunchInstanceAWSTaskArgs) e
 
 	return nil
 }
+
+func FetchInstancesDescriptionAWS(ctx context.Context, args *LaunchInstanceAWSTaskArgs) error {
+	logger := ctxval.Logger(ctx)
+	logger.Debug().Msg("Started fetch instances description")
+
+	// skip job if reservation already contains errors
+	err := checkExistingError(ctx, args.ReservationID)
+	if err != nil {
+		return fmt.Errorf("step skipped: %w", err)
+	}
+
+	updateStatusBefore(ctx, args.ReservationID, "Fetching instance(s) description")
+	defer updateStatusAfter(ctx, args.ReservationID, "Instance(s) description fetched", 1)
+
+	rDao := dao.GetReservationDao(ctx)
+	instances, err := rDao.ListInstances(ctx, args.ReservationID)
+	if err != nil {
+		return fmt.Errorf("cannot get instances list: %w", err)
+	}
+	instancesIDList := make([]string, len(instances))
+	for i, instance := range instances {
+		instancesIDList[i] = instance.InstanceID
+	}
+	ec2Client, err := clients.GetEC2Client(ctx, args.ARN, args.Region)
+	if err != nil {
+		return fmt.Errorf("cannot create new ec2 client from config: %w", err)
+	}
+	backoffInterval := [...]int64{1000, 500, 500, 1000, 2000}
+	for _, interval := range backoffInterval {
+		time.Sleep(time.Duration(interval) * time.Millisecond)
+		instancesDescriptionList, err := ec2Client.DescribeInstanceDetails(ctx, instancesIDList)
+		if err != nil {
+			return fmt.Errorf("cannot get list instances description: %w", err)
+		}
+		if len(instancesDescriptionList) == 0 {
+			continue
+		}
+		for _, instance := range instancesDescriptionList {
+			err := rDao.UpdateReservationInstance(ctx, args.ReservationID, instance)
+			if err != nil {
+				return fmt.Errorf("cannot update instance description: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
```

Comment:
Note for me or @ezr-ondrej - the https://github.com/RHEnVision/provisioning-backend/pull/461 requires all functions to use the new `nilUnlessTimeout` here so if this gets merged before this needs to be updated.